### PR TITLE
PRO-1284 delete permanently

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: modules/@apostrophecms/doc-type/index.js
Comment (boutell) [inline modules/@apostrophecms/doc-type/index.js:181]: Unrelated fix for deprecation warning.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:27]: The "More" menu now appears in more circumstances because "Delete" is available in the doc editor even when archived, so I added more guards against offering other options at inappropriate times.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:166]: I renamed this because "restore" is no longer the only thing you can do to an archived item in the editor; you can also delete it permanently.

**Review by abea — CHANGES_REQUESTED** [review]
We get a server error when opening a doc for editing while arcived. It's looking for the published mode of the doc and not finding it, I think.

```
GET /api/v1/article/ckooibnvi000ss83rrqjzsegl:en:draft?aposMode=published:

notfound: notfound
    at Object.error (/Users/abea/Apos/apostrophe/modules/@apostrophecms/error/index.js:29:23)
    at getOne (/Users/abea/Apos/apostrophe/modules/@apostrophecms/piece-type/index.js:190:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/abea/Apos/apostrophe/modules/@apostrophecms/module/index.js:152:30
GET /api/v1/article/ckooibnvi000ss83rrqjzsegl:en:draft?archived=any&aposMode=published:

notfound: notfound
    at Object.error (/Users/abea/Apos/apostrophe/modules/@apostrophecms/error/index.js:29:23)
    at getOne (/Users/abea/Apos/apostrophe/modules/@apostrophecms/piece-type/index.js:190:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/abea/Apos/apostrophe/modules/@apostrophecms/module/index.js:152:30
```

On delete there's another server error when it tries to PATCH the draft:

```
PATCH /api/v1/article/ckooibnvi000ss83rrqjzsegl:en:draft?aposMode=draft:

notfound: notfound
    at Object.error (/Users/abea/Apos/apostrophe/modules/@apostrophecms/error/index.js:29:23)
    at /Users/abea/Apos/apostrophe/modules/@apostrophecms/piece-type/index.js:655:29
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Object.withLock (/Users/abea/Apos/apostrophe/modules/@apostrophecms/lock/index.js:184:18)
    at async /Users/abea/Apos/apostrophe/modules/@apostrophecms/module/index.js:152:30
```

That's only in the editor modal. The manager modal experience seems fine.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManagerDisplay.vue
Comment (abea) [inline modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManagerDisplay.vue:101]: this isn't in the `emits` array below.

**Review by abea — CHANGES_REQUESTED** [review]
one other little bit.

**Comment by boutell** [comment]
Keeping this one on hold until PRO-1304 lands. Then I'll adjust it to suit the new situation.

**Comment by boutell** [comment]
Leaving the branch for now in case I decide I do want to reference it.

**Pull request closed without merging**
